### ISSUE TITLE
Error 500 instead of 400 for invalid UUID in POST body

### ISSUE THREAD

**Issue by guilliamxavier** [api-platform/core#4189]
**API Platform version(s) affected**: 2.6.3, 2.6.4
- Symfony 4.4
- `ramsey/uuid` 3.9.3

**Description**  
(see title) when using the `UuidInterface` type declaration for automatic denormalization

**How to reproduce**  
On PHP 7.4
```php
<?php

namespace App\Entity;

use ApiPlatform\Core\Annotation\ApiResource;
use Ramsey\Uuid\UuidInterface;

/** @ApiResource */
class Foo
{
    public int $id = 0;
    public ?UuidInterface $uuid = null;
    public ?\DateTimeInterface $datetime = null;
}
```

Request: `POST /foos` with body (Content-Type: application/ld+json):
```json
{"uuid": "foo"}
```

Expected: 400 Bad Request
Actual: 500 Internal Server Error
(message: "Invalid UUID string: foo")

**Possible Solution**  
In [`ApiPlatform\Core\Bridge\RamseyUuid\Serializer\UuidDenormalizer`](https://github.com/api-platform/core/blob/2.6/src/Bridge/RamseyUuid/Serializer/UuidDenormalizer.php), wrap `return Uuid::fromString($data);` with a try-catch (like in [`ApiPlatform\Core\Bridge\RamseyUuid\Identifier\Normalizer\UuidNormalizer`](https://github.com/api-platform/core/blob/2.6/src/Bridge/RamseyUuid/Identifier/Normalizer/UuidNormalizer.php)) and convert the `Ramsey\Uuid\Exception\InvalidUuidStringException` to an appropriate exception (a subclass of `Symfony\Component\Serializer\Exception\ExceptionInterface`)?

**Additional Context**  
For this body:
```json
{"datetime": "foo"}
```
I get a 400 Bad Request, as expected
(message: "DateTimeImmutable::__construct(): Failed to parse time string (foo) at position 0 (f): The timezone could not be found in the database")

**Comment by soyuka** [comment]
Isn't that related to https://github.com/api-platform/core/issues/2191?

**Comment by guilliamxavier** [comment]
@soyuka: I don't think so (this has nothing to do with "short UUID"). Rather #3045 (but that was for e.g. `GET /foos/foo`) and #3473 (but that was for `ramsey/uuid` ^4.0).

By the way, this issue also exists when using the `$uuid` as the id, i.e. annotating it with `@ApiProperty(identifier=true)`

**Comment by guilliamxavier** [comment]
Fixed in #4200, to be released in 2.6.5
